CampusTour/ViewControllers/SearchViewController.swift

```diff
@@ -8,97 +8,63 @@
 
 import UIKit
 
-class SearchViewController: UIViewController, UITableViewDelegate, UITableViewDataSource, UISearchBarDelegate {
+class SearchViewController: UIViewController, UITableViewDelegate, UITableViewDataSource, FilterFunctionsDelegate, PopupFilterProtocol {
     
-    var searchController: UISearchController!
+    var searchBar: UISearchBar!
     var searchResultsTableView: UITableView!
     let itemFeedViewController = ItemFeedViewController()
     var filterBar: FilterBar!
     var arButton: UIBarButtonItem!
+    
+    //Replace with data from DataManager
+    let allData = [Any]()
+    var popupViewController: PopupViewController!
+    private let cellReuse = "reuse"
+    private var currentModalMode: Filter?
+    private var filterBarCurrentStatus = FilterBarCurrentStatus(Filter.general.rawValue, Filter.date.rawValue)
+    private var blackView: UIView!
+    
+    private var isModal = false {
```

Comment:
what does isModal do?

Reply:
It checks whether the modal view is already visible, and if it is, it removes it. otherwise, it displays it. It’s there since the filter buttons call one function